CLDR-18461 Update ISO languages for v48

conradarcturus opened the pull request:
I was building a new dataset with ISO codes and I thought "I should check to make sure the CLDR ISO codes are up to date". Turns out there are a few changes.

Followed the protocol in this document:

https://cldr.unicode.org/development/updating-codes/update-languagescriptregion-subtags

Question: @btangmu's previous PR added the languages to the XMLs but it doesn't look like the script did that for him -- do I have to do that manually? https://github.com/unicode-org/cldr/pull/3700

CLDR-18461

- [x] This PR completes the ticket.

<!--
Thank you for your pull request.
Please see https://cldr.unicode.org/index/process for general
information on contributing to CLDR.

1. Make sure the ticket is filed at
https://unicode-org.atlassian.net/projects/CLDR/
2. Update the PR title and first line of this
message to include the ticket ID (CLDR-_____)
3. You will be automatically asked to sign the contributors’
license before the PR is accepted.
- sign: https://cla-assistant.io/unicode-org/cldr
- license: https://www.unicode.org/copyright.html#License
-->

ALLOW_MANY_COMMITS=true

srl295 commented:
Yes, I meant to mention that the language sub tag registry had been updated

btangmu commented:
@conradarcturus Looking at my notes, I see that I ran GenerateValidityXML as part of https://github.com/unicode-org/cldr/pull/3700 . Any changes to I made to XML files are more likely to be from running tools than from hand-editing, though I don't remember for sure.

conradarcturus commented:
Perfect! That's the script that ran it. 2 questions though:

* The guide mentions updating en.xml --- uhhh we haven't added new languages or scripts to en.xml. In fact, we only have 684 localeDisplayNames (out of 7963) and 55 variant names (out of 122 variants).
* The last ISO update was Jan 15 -- based on their cadence I wouldn't be surprised if there is another update April 15 -- do you know if there is an impending update?

srl295 commented on tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/language-subtag-registry:
```diff
@@ -1,4 +1,4 @@
-File-Date: 2024-06-14
+File-Date: 2025-03-10
```
ok good - didn't see this at first

conradarcturus commented:
Turns out because of the deprecated codes I need to actually do a few more changes anyway, not ready yet

conradarcturus commented:
Added language aliases for the deprecated codes. Also added Language Matches since a different document requested that update when we add new languages to macrolanguages.

conradarcturus commented on tools/cldr-code/src/main/java/org/unicode/cldr/tool/GenerateLikelySubtags.java:
```diff
@@ -499,6 +499,9 @@ public static void main(String[] args) throws IOException {
                                 {"sga", "sga_Latn_IE"},
+
+                                // Perserve data now that dek has been merged with sqm
+                                {"sqm", "sqm_Latn_CF"},
```
If we didn't do this -- then our import of SIL data was going to merge "dek_Latn_CM" with "sqm_Latn_CF" into "sqm_Latn_CM" -- however there are <5000 speakers in Cameroon and 50,000 speakers in CAF - so I needed to add an override here. I can add more comprehensive Cameroon and CAF data later -- already this PR is doing a lot...

macchiati reviewed (approved):
Looks great, thanks!

macchiati commented:
> The guide mentions updating en.xml --- uhhh we haven't added new languages or scripts to en.xml. In fact, we only have 684 localeDisplayNames (out of 7963) and 55 variant names (out of 122 variants).

The guide is out of date; we have never tried to add all the English language / script names to en.xml. We might do that along with the endonyms, but in another directory, not main — see the endonym ticket.
And we have only added a subset of variants (and even those there it a ticket to reduce those to the ones in more common usage). Again, we might add them all along, but not in main.
